app/controllers/admin/payment_confirmation_report_uploads_controller.rb

```diff
@@ -0,0 +1,20 @@
+module Admin
+  class PaymentConfirmationReportUploadsController < BaseAdminController
+    before_action :ensure_service_operator
+
+    def new
+      @payroll_run = PayrollRun.find(params[:payroll_run_id])
+    end
+
+    def create
+      @payroll_run = PayrollRun.find(params[:payroll_run_id])
+      @payment_confirmation = PaymentConfirmation.new(@payroll_run, params[:file])
+      if @payment_confirmation.ingest
+        @payroll_run.update!(confirmation_report_uploaded_by: admin_session.user_id)
```

Comment:
It definitely makes sense. I've made the change, but I'm not sure how I would test this, or whether I need to. I don't see any easy way to make the `update!` call fail.

Reply:
Hmmm, good question! Making `update!` fail with this call will be hard as it will only fail if the validations fail or if the write to the database fails. Neither is something that we can trigger without faking something lower in the stack. Given that I think it's ok to just extract the code and wrap in a transaction without specific test coverage for it.